### DISCUSSION TITLE
How to get the same x,y,z keypoint values using 1.x version?

### DISCUSSION THREAD

**Discussion by jhkim-lumanlab** [open-mmlab/mmpose#2549] [Q&A] [answered]
First of all, thank you for the great works!

In the past, I did hand keypoint detection work using version 0.x of your repository. I used the config file 'configs/hand/2d_kpt_sview_rgb_img/topdown_heatmap/coco_wholebody_hand/hrnetv2_w18_coco_wholebody_hand_256x256_dark.py' in combination with the provided checkpoint. This allowed me to obtain x, y, z values for each keypoint.

Recently, I've attempted to utilize version 1.x, and found that the config file 'configs/hand_2d_keypoint/topdown_heatmap/coco_wholebody_hand/td-hm_hrnetv2-w18_dark-8xb32-210e_coco-wholebody-hand-256x256.py' uses the exact same checkpoint. However, the inference results only include x, y values for each keypoint, rather than the x, y, z values I was able to obtain before.  

I would like to continue obtaining x, y, z values for each keypoint. Could you please guide me on how to achieve this, even when using the new version?

**Comment by Ben-Louis** [comment]
The method "hrnetv2_w18_coco_wholebody_hand_256x256_dark" is used to detect 2D keypoints of the hand and does not provide a z-value.

**Reply by jhkim-lumanlab** [reply]
Could you provide me with information about the third value in each keypoint array? 

For example, when I use the inference_top_down_pose_model function with the following parameters:
inference_top_down_pose_model(
    pose_model,
    img_path,
    person_results,
    bbox_thr=0.7,
    format='xyxy',
    dataset=dataset,
    dataset_info=dataset_info
)
I receive the following return values:

([{
    'bbox': array([182.84047   , 350.2889    , 268.3354    , 459.6789    ,
           0.98880297], dtype=float32),
    'keypoints': array([[1.91999542e+02, 3.85606201e+02, 1.49724647e-01],
           [2.15127289e+02, 3.87819122e+02, 1.96995884e-01],
           [2.37886658e+02, 3.90352020e+02, 2.64756858e-01],
           ...
           [2.15984985e+02, 4.39371796e+02, 2.61255413e-01]], dtype=float32)}
 ],
 [{}])
 
The pose model configuration is:
 pose_model = init_pose_model(pose_config, pose_checkpoint)
pose_config = '/home/jhkim/mmdet/mmpose/configs/hand/2d_kpt_sview_rgb_img/topdown_heatmap/coco_wholebody_hand/hrnetv2_w18_coco_wholebody_hand_256x256_dark.py'
pose_checkpoint = '/home/jhkim/mmdet/mmpose/checkpoints/hrnetv2_w18_coco_wholebody_hand_256x256_dark-a9228c9c_20210908.pth'

I thought the third value is the z-value but clearly it is not.

**Reply by Ben-Louis** [reply, marked as answer]
Hello @jhkim-lumanlab, I wanted to clarify that the third value displayed is actually the confidence of the prediction. In the new version, this value is now assigned to the `keypoint_scores` attribute.

**Reply by jhkim-lumanlab** [reply]
Thank you for the information:)
